goframe-gf: init at 2.9.3

cococolanosugar opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ x ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [2.9.3 Release Notes](https://github.com/gogf/gf/releases/tag/v2.9.3) 
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+
+    license = licenses.mit;
+
+    maintainers = with maintainers; [ "cococolanosugar" ];
```
```suggestion
    maintainers = with lib.maintainers; [ cococolanosugar ];
```
Maintainers must be declared in [`maintainers/maintainer.list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix). Please follow the [Maintainers' README](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) to add your entry there.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+    GOWORK = "off";
+  };
+
+  meta = with lib; {
```
```suggestion
  meta = {
```
Nit: Avoid [wide `with` scopes](https://nix.dev/guides/best-practices#with-scopes) in attribute sets. It's alright to repeat `lib.*` when you need it:
```nix
{
  meta = {
    license = lib.licenses.publicDomain;
    maintainers = with lib.maintainers; [ maintainer ];
  };
}
```

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+  };
+
+  meta = with lib; {
+
```
```suggestion
```
For consistency with other packages, avoid adding so many newlines between attributes.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+{
+
+  lib,
+
+  buildGoModule,
+
+  fetchFromGitHub,
+
+}:
```
```suggestion
{
  lib,
  buildGoModule,
  fetchFromGitHub,
}:
```
Ditto for the function header.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+  version = "2.9.0";
+
+  src = fetchFromGitHub {
+    owner = "cococolanosugar";
```
```suggestion
    owner = "gogf";
```
Please follow upstream's repo, not your fork.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+  src = fetchFromGitHub {
+    owner = "cococolanosugar";
+    repo = "gf";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

There is a chance that upstream will name a tag with the same name as a `git` HEAD, which can cause issues with this fetcher's fixed output derivation if upstream pushes to that branch. Since we want the tags, which are usually less prone to change, we can use the `tag` attribute that automatically prefixes the value with `refs/tags`, preventing the fetcher from fetching anything that's not a tag.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+    "-w"
+  ];
+
+  doCheck = false;
```
Please re-enable checks, or add a comment saying why they can't run for this package.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+
+    maintainers = with maintainers; [ "cococolanosugar" ];
+
+    mainProgram = pname;
```
```suggestion
    mainProgram = "gf";
```
Avoid interpolating the `pname` attribute. Ideally, a downstream user can change the `pname` attribute using `overrideAttrs` without altering any other attributes of the derivation.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+
+    description = "A powerful framework for faster, easier, and more efficient project development.";
+
+    homepage = "https://github.com/gogf/gf";
```
```suggestion
    homepage = "https://goframe.org";
```

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+
+  meta = with lib; {
+
+    description = "A powerful framework for faster, easier, and more efficient project development.";
```
Please observe the general requirements for the `meta.description` field listed in the [Nixpkgs Documentation](https://nixos.org/manual/nixpkgs/unstable/#var-meta-description).

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,51 @@
+  subPackages = [ "." ];
+  modRoot = "cmd/gf";
+
+  vendorHash = null;
```
I don't see a `vendor` directory in the upstream repository. Are you sure this compiles when set to `null`?

SigmaSquadron reviewed (changes requested):
Welcome to Nixpkgs! Thanks for packaging this software.

Please review our [Commit Conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) to squash your intermediate/duplicate/fixup commits.

I've left some additional review comments below.

hugolgst reviewed (changes requested):
you should run an interactive rebase to squash all the commits with `git rebase -i origin/master` into one
then you can force push to your branch `git push --force-with-lease`

hugolgst reviewed (changes requested):
actually you should have two commits, one for adding you as a maintainer and one for initialising the derivation

hugolgst commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,41 @@
+  ];
+
+  meta = {
+    description = "gf is a powerful CLI tool for building GoFrame application with convenience.";
```
```suggestion
    description = "gf is a powerful CLI tool for building GoFrame application with convenience";
```

cococolanosugar commented:
It seems that the GoFrame team does not maintain a correct and complete go.sum file for the gf CLI. 
I’ve already created an issue for it, and I’ll try again once they’ve resolved the problem.

cococolanosugar committed:
goframe-cli: init at 2.9.3

goframe-cli: init at 2.9.3

Update pkgs/by-name/go/goframe-cli/package.nix

Co-authored-by: Fernando Rodrigues <alpha@sigmasquadron.net>

goframe-cli: init at 2.9.3

Merge branch 'NixOS:master' into feature/goframe-cli

goframe-cli: init at 2.9.3

Update pkgs/by-name/go/goframe-cli/package.nix

Co-authored-by: Fernando Rodrigues <alpha@sigmasquadron.net>

goframe-gf: init at 2.9.3

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,41 @@
+}:
+
+buildGoModule rec {
+  pname = "gf";
```
The attribute name should match the `pname` if possible. Please review the [package naming guidelines](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming):
- It _should_ be identical to the upstream package name.
- It _must_ not contain uppercase letters.

cococolanosugar replied:
This CLI tool is originally named gf, but since there is already a package named gf in nixpkgs, I have renamed the package to goframe-gf while keeping the binary name as gf.

SigmaSquadron commented on pkgs/by-name/go/goframe-cli/package.nix:
```diff
@@ -0,0 +1,41 @@
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
```
```suggestion
buildGoModule (finalAttrs: {
```

I believe the `buildGoModule` builder now supports the `finalAttrs` pattern.

SigmaSquadron reviewed (commented):
Looks good overall.

SigmaSquadron commented:
Please review our [Commit Conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions) to squash your intermediate/duplicate/fixup commits.

SigmaSquadron reviewed (changes requested):
argh, misclick.
